tests/usbredir_test.go

```diff
@@ -0,0 +1,177 @@
+			tests.WaitForSuccessfulVMIStart(vmi)
+		})
+
+		FIt("should fail to connect to VMI's usbredir socket", func() {
+			usbredirVMI, err := virtClient.VirtualMachineInstance(vmi.ObjectMeta.Namespace).USBRedir(vmi.ObjectMeta.Name)
+			Expect(err).To(HaveOccurred())
+			Expect(usbredirVMI).To(BeNil())
+		})
+	})
+
+	Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component] A VirtualMachineInstance with usbredir support", func() {
+		tests.BeforeAll(func() {
+			virtClient, err = kubecli.GetKubevirtClient()
+			util.PanicOnError(err)
+
+			tests.BeforeTestCleanup()
+			vmi = tests.NewRandomVMI()
+			vmi.Spec.Domain.Devices.ClientPassthrough = &v1.ClientPassthroughDevices{}
+			Expect(virtClient.RestClient().Post().Resource("virtualmachineinstances").Namespace(util.NamespaceTestDefault).Body(vmi).Do(context.Background()).Error()).To(Succeed())
+			tests.WaitForSuccessfulVMIStart(vmi)
+		})
+
+		Context("with an usbredir connection", func() {
+
+			usbredirConnect := func(connStop chan struct{}) {
+				pipeInReader, pipeInWriter := io.Pipe()
+				pipeOutReader, pipeOutWriter := io.Pipe()
+				defer pipeInReader.Close()
+				defer pipeOutReader.Close()
+
+				k8ResChan := make(chan error)
+				readStop := make(chan []byte)
+
+				By("Stablishing communication with usbredir socket from VMI")
+				go func() {
+					defer GinkgoRecover()
+					usbredirVMI, err := virtClient.VirtualMachineInstance(vmi.ObjectMeta.Namespace).USBRedir(vmi.ObjectMeta.Name)
+					if err != nil {
+						k8ResChan <- err
+						return
+					}
+
+					k8ResChan <- usbredirVMI.Stream(kubecli.StreamOptions{
+						In:  pipeInReader,
+						Out: pipeOutWriter,
+					})
+				}()
+
+				By("Exchanging hello message between client and QEMU's usbredir")
+				go func() {
+					defer GinkgoRecover()
+					buf := make([]byte, 1024, 1024)
+
+					// write hello message to remote (VMI)
+					nw, err := pipeInWriter.Write(helloMessageLocal)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(nw).To(Equal(len(helloMessageLocal)))
+
+					// reading hello message from remote (VMI)
+					nr, err := pipeOutReader.Read(buf)
+					if err != nil && err != io.EOF {
+						Expect(err).ToNot(HaveOccurred())
+						return
+					}
+					if nr == 0 && err == io.EOF {
+						log.Log.Info("zero bytes read from usbredir socket.")
+						return
+					}
+					readStop <- buf[0:nr]
+				}()
+
+				select {
+				case response := <-readStop:
+					By("Checking the response from QEMU's usbredir")
+					// Comparing the actual messages could be error prone due the fact that:
+					// 1. Part of the return value is a qemu release version, e.g: 5.2.0 (test would break with different release!)
+					// 2. Capabilities can change over time which means the message would be different then the one hardcoded, correct nonetheless.
+					// I'm keeping the helloMessageRemote to have a proof of working example that could also be used if needed.
+					Expect(len(response)).To(Equal(len(helloMessageRemote)))
```

Comment:
Further to my previous comment, you can even do something like this for more clarity:
```
Expect(response).ToNot(BeEmpty(), "response should not be empty")
Expect(len(response)).To(Equal(len(helloMessageRemote)))
```